File: backend/src/main/java/ru/focsit/backend/dto/SignUpRequest.java
package ru.focsit.backend.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class SignUpRequest {

    @NotBlank(message = "Имя пользователя не может быть пустыми")
    private String username;

    @NotBlank(message = "Адрес электронной почты не может быть пустыми")
    @Email(message = "Email адрес должен быть в формате user@example.com")
    private String email;

    private String password;

    private String role;
}

File: mobile/app/src/main/java/ru/focsit/mobile/data/auth/SignUpRequest.kt
package ru.focsit.mobile.data.auth

/**
 * Запрос на регистрацию пользователя в системе.
 * Этот класс используется для отправки данных о пользователе при регистрации в системе.
 *
 * @property username Имя пользователя, которое будет использоваться для входа в систему.
 * @property email Электронная почта пользователя, которая будет использована для связи и восстановления пароля.
 * @property password Пароль пользователя для доступа к учетной записи.
 * @property role Роль пользователя в системе (например, "пользователь", "админ").
 */
data class SignUpRequest(
    val username: String,
    val email: String,
    val password: String,
    val role: String
)